src/Modules/Shared/PowerShellGet/PSModule.psm1

```diff
@@ -19,7 +19,7 @@ function IsLinux { $PSVariable = Get-Variable -Name IsLinux -ErrorAction Ignore;
 function IsOSX { $PSVariable = Get-Variable -Name IsOSX -ErrorAction Ignore; return ($PSVariable -and $PSVariable.Value) }
 function IsCoreCLR { $PSVariable = Get-Variable -Name IsCoreCLR -ErrorAction Ignore; return ($PSVariable -and $PSVariable.Value) }
 
-if(IsWindows)
+if((IsWindows) -and (-not (IsCoreCLR)))

```

Comment:
@jianyunt  On Nano Server, IsCoreCLR returns $false, so this doesn't change the installation location on Nano Server.